modules/pensions/app/controllers/pensions/v0/claims_controller.rb

```diff
@@ -67,6 +71,30 @@ def create
 
       private
 
+      # Build payload and submit to EventBusSubmissionJob
+      #
+      # @param claim [Pensions::SavedClaim]
+      def submit_traceability_to_event_bus(claim)
+        Kafka::EventBusSubmissionJob.perform_async(
+          'submission_trace_mock_dev',
+          {
+            'data' => {
+              'ICN' => user_icn,
+              'currentID' => claim&.confirmation_number.to_s,
+              'submissionName' => Pensions::FORM_ID,
+              'state' => Kafka::State::RECEIVED
+            }
+          }
+        )
+      end
+
+      # Retrieves the Integration Control Number (ICN) associated with the current user
+      #
+      # @return [String] The ICN of the current user, or an empty string if not available
+      def user_icn
+        (current_user&.icn || UserAccount.find_by(id: current_user&.user_account_uuid)&.icn).to_s
```

Comment:
Is `current_user` an instance of `UserAccount`? If yes, then I don't know if we need the OR statement since `current_user`, if present, would have the `icn` attribute.